appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/MetricsService.java

```diff
@@ -418,15 +419,16 @@ private void deregisterApplication(String applicationName) {
     public String getApplicationName() {
         InvocationManager invocationManager = Globals.getDefaultBaseServiceLocator()
                 .getService(InvocationManager.class);
-        if (invocationManager.getCurrentInvocation() == null) {
```

Comment:
NB: while the use of thread local should ensure that multiple calls to `getCurrentInvocation()` should return the same instance it is better to not take changes and do it once. Also improves readability.